include/exadg/acoustic_conservation_equations/spatial_discretization/operators/grad_div_operator.h

```diff
@@ -0,0 +1,537 @@
+    {
+      do_evaluate_formulation<FormulationGradientDivergenceTerms::SkewSymmetric>(dst,
+                                                                                 src,
+                                                                                 time,
+                                                                                 zero_dst_vector);
+    }
+    else
+    {
+      AssertThrow(false, dealii::ExcMessage("Not implemented."));
+    }
+  }
+
+  template<FormulationGradientDivergenceTerms formulation>
+  void
+  do_evaluate_formulation(BlockVectorType &       dst,
+                          BlockVectorType const & src,
+                          double const            time,
+                          bool const              zero_dst_vector) const
+  {
+    evaluation_time = (Number)time;
+
+    matrix_free->loop(&This::cell_loop<formulation>,
+                      &This::face_loop<formulation>,
+                      &This::boundary_face_loop<formulation>,
+                      this,
+                      dst,
+                      src,
+                      zero_dst_vector,
+                      dealii::MatrixFree<dim, Number>::DataAccessOnFaces::values,
+                      dealii::MatrixFree<dim, Number>::DataAccessOnFaces::values);
+  }
+
+  template<FormulationGradientDivergenceTerms formulation>
+  void
+  do_cell_integral(CellIntegratorP & pressure,
+                   CellIntegratorU & velocity,
+                   Number const      rho_inv,
+                   Number const      rhocc) const
+  {
+    for(unsigned int q : pressure.quadrature_point_indices())
+    {
+      if constexpr(formulation == FormulationGradientDivergenceTerms::Weak)
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        scalar const flux_mass     = kernel.get_volume_flux_weak_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_divergence(rho_inv * flux_mass, q);
+      }
+      else if constexpr(formulation == FormulationGradientDivergenceTerms::Strong)
+      {
+        scalar const flux_momentum = kernel.get_volume_flux_strong_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_value(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+      else if constexpr(formulation == FormulationGradientDivergenceTerms::SkewSymmetric)
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+      else
+      {
+        AssertThrow(false, dealii::ExcMessage("Not implemented."));
+      }
+    }
```

Comment:
Here, it would be very easy (i.e. with almost no code duplication) to write the for-loop inside the if-else-branches. Then, one would probably make the current motivation for `constexpr` and thte use of a template argument (for performance reasons) superfluous.

Reply:
I agree, but it doesn't work for the face loop like this :/